### PR TITLE
add new website

### LINKED ISSUE #996: New website
Hello everyone,

Since the current website could still need some improvements, I've made a new website for this project. Let me know what you think about it. The newsletter thing still isn't integrated, so that's why I didn't open a pr and it currently ist still a bit broken on mobile.

https://github.com/maltejur/disease.sh
https://maltejur.github.io/disease.sh/

### PR DESCRIPTION
**Pull request by maltejur**
See #1001 and #996. A bot somehow messed up the last pr, so I am opening a new one.

closes #996

### CODE REVIEW

**Comment by puf17640** [comment]
closes #996

**Comment by maltejur** [comment]
As commented on the other pr, a new recaptcha v3 would need to be generated and updated [here](https://github.com/maltejur/API/blob/fcdbfffc70ef08ef490af4473116b8c1817ea4c1/public/assets/js/index.js#L228) and [here](https://github.com/maltejur/API/blob/fcdbfffc70ef08ef490af4473116b8c1817ea4c1/public/assets/js/index.js#L224) in the pr, as well as in the production environment variables. Asides from that, everything should be ready

**Comment by puf17640** [comment]
I just replied on the closed PR (pretty dumb thing now that I think of it haha)

**Comment by maltejur** [comment]
Okay now you have a environment variable `CAPTCHA_TOKEN` for the public captcha token

**Comment by ebwinters** [comment]
@puf17640 Giving 

@maltejur 6Ld6odEaAAAAAJhhaLm_cQffoNQr4H4mLszvLIaa is the new public key

@puf17640 I'll have to PM coleman the new private key

**Comment by maltejur** [comment]
@ebwinters I can't do anything with the public key since you now configure it in the same place as the private key

**Comment by ebwinters** [comment]
> @ebwinters I can't do anything with the public key since you now configure it in the same place as the private key

you need the new public key to update the PR in those two places you marked

**Comment by maltejur** [comment]
https://github.com/disease-sh/API/pull/1001#issuecomment-836748818

It is now the environment variable `CAPTCHA_TOKEN`

**Comment by dontlaugh** [comment]
@ebwinters missed your DM this morning. Try me again to send the private key. 

Have the **names** of any config variables changed? This will require an update to the Kubernetes deployment manifest (yaml) on our end.

I'll post here when it's safe to merge this.

**Comment by ebwinters** [comment]
@maltejur let's keep it how you have it for the PR. I'll let coleman know the Private key to add to deploy. But, won't merge this yet, I want to test

**Comment by dontlaugh** [comment]
This deployment will need to wait at least a day. My colleague is not around, and to avoid downtime, I'd really like her by my side to babysit this. I'm probably just paranoid, but today is kind of busy, and I want to make sure cool change goes through without a hitch.

I'll get back to you all tomorrow. :heart:

**Comment by maltejur** [comment]
> Have the **names** of any config variables changed? This will require an update to the Kubernetes deployment manifest (yaml) on our end.

No changes of existing variables, just the new `CAPTCHA_TOKEN` for the captcha public key.

File: .eslintrc.json [hunk 1]
```diff
@@ -13,5 +15,12 @@
...
-	}
+	},
+	"parserOptions": {
+        "sourceType": "module"
+	},
+    "env": {
+        "browser": true,
+        "node": true
+    }
 }
```
Comment (Panquesito7) [inline .eslintrc.json:26]: Maybe not related to the PR, but this file is missing an endline.
```suggestion
}

```

**Comment by dontlaugh** [comment]
From our end - Kait is still out. I think we will need to wait until Monday. Sorry folks!

**Comment by dontlaugh** [comment]
We have created a new secret on our end (not overwriting the old, just to be safe). This secret is ready to be applied to the existing deployments. Our diff for _deployment_ objects:

<summary>diff</summary>
<details>

```diff
diff --git a/apis/novelcovid/novelcovid.api.deployment.yaml b/apis/novelcovid/novelcovid.api.deployment.yaml
index 13c0142..f45af67 100644
--- a/apis/novelcovid/novelcovid.api.deployment.yaml
+++ b/apis/novelcovid/novelcovid.api.deployment.yaml
@@ -45,12 +45,12 @@ spec:
         - name: MAILGUN_API_KEY
           valueFrom:
             secretKeyRef:
-              name: api-secret
+              name: api-secret-new
               key: MAILGUN_API_KEY
         - name: CAPTCHA_SECRET
           valueFrom:
             secretKeyRef:
-              name: api-secret
+              name: api-secret-new
               key: CAPTCHA_SECRET
         image: novelcovid/novelcovid-api:sha-14aa3e8
         imagePullPolicy: "Always"
diff --git a/apis/novelcovid/novelcovid.scraper.deployment.yaml b/apis/novelcovid/novelcovid.scraper.deployment.yaml
index 663dbb3..0b41612 100644
--- a/apis/novelcovid/novelcovid.scraper.deployment.yaml
+++ b/apis/novelcovid/novelcovid.scraper.deployment.yaml
@@ -43,12 +43,12 @@ spec:
         - name: MAILGUN_API_KEY
           valueFrom:
             secretKeyRef:
-              name: api-secret
+              name: api-secret-new
               key: MAILGUN_API_KEY
         - name: CAPTCHA_SECRET
           valueFrom:
             secretKeyRef:
-              name: api-secret
+              name: api-secret-new
               key: CAPTCHA_SECRET
         image: novelcovid/novelcovid-api:sha-14aa3e8
         imagePullPolicy: "Always"
diff --git a/apis/novelcovid/update-checker.cronjob.yaml b/apis/novelcovid/update-checker.cronjob.yaml
index 53e3ae9..60f0ac3 100644
--- a/apis/novelcovid/update-checker.cronjob.yaml
+++ b/apis/novelcovid/update-checker.cronjob.yaml
@@ -38,12 +38,13 @@ spec:
             - name: MAILGUN_API_KEY
               valueFrom:
                 secretKeyRef:
-                  name: api-secret
+                  name: api-secret-new
                   key: MAILGUN_API_KEY
+            # TODO can we get rid of CAPTCHA for this?
             - name: CAPTCHA_SECRET
               valueFrom:
                 secretKeyRef:
-                  name: api-secret
+                  name: api-secret-new
                   key: CAPTCHA_SECRET
             - name: UPDATE_WEBHOOK
               valueFrom:

```


</details>

**Comment by dontlaugh** [comment]
We are ready on our end. Here is the sequencing of events. It will take some coordination.

### Deploy

1. You merge the website update (this PR)
2. I immediately merge our flux gitops repo change, which includes the new secret

There will be a window of downtime while things restart, and k8s hopefully coordinates all that

Cloudflare caching should keep website users mostly happy.

### Rollback Plan

If the deployment/code breaks for any reason, I will turn off autodeploy in flux, so subsequent merges to API repo will not be deployed.

Then I will manually put the old image back. and point it at the old secret. This should restore the old website.

----
cc @kaitmore @hiromis
If there's anything else I should think of, lemme know. :+1: 

Just ping me on Discord when you're ready to go. Best if it's during business hours :smiling_face_with_three_hearts:

**Review by ebwinters — APPROVED** [review]

**Comment by maltejur** [comment]
> diff
> 
> ```diff
> diff --git a/apis/novelcovid/novelcovid.api.deployment.yaml b/apis/novelcovid/novelcovid.api.deployment.yaml
> index 13c0142..f45af67 100644
> --- a/apis/novelcovid/novelcovid.api.deployment.yaml
> +++ b/apis/novelcovid/novelcovid.api.deployment.yaml
> @@ -45,12 +45,12 @@ spec:
>          - name: MAILGUN_API_KEY
>            valueFrom:
>              secretKeyRef:
> -              name: api-secret
> +              name: api-secret-new
>                key: MAILGUN_API_KEY
>          - name: CAPTCHA_SECRET
>            valueFrom:
>              secretKeyRef:
> -              name: api-secret
> +              name: api-secret-new
>                key: CAPTCHA_SECRET
>          image: novelcovid/novelcovid-api:sha-14aa3e8
>          imagePullPolicy: "Always"
> diff --git a/apis/novelcovid/novelcovid.scraper.deployment.yaml b/apis/novelcovid/novelcovid.scraper.deployment.yaml
> index 663dbb3..0b41612 100644
> --- a/apis/novelcovid/novelcovid.scraper.deployment.yaml
> +++ b/apis/novelcovid/novelcovid.scraper.deployment.yaml
> @@ -43,12 +43,12 @@ spec:
>          - name: MAILGUN_API_KEY
>            valueFrom:
>              secretKeyRef:
> -              name: api-secret
> +              name: api-secret-new
>                key: MAILGUN_API_KEY
>          - name: CAPTCHA_SECRET
>            valueFrom:
>              secretKeyRef:
> -              name: api-secret
> +              name: api-secret-new
>                key: CAPTCHA_SECRET
>          image: novelcovid/novelcovid-api:sha-14aa3e8
>          imagePullPolicy: "Always"
> diff --git a/apis/novelcovid/update-checker.cronjob.yaml b/apis/novelcovid/update-checker.cronjob.yaml
> index 53e3ae9..60f0ac3 100644
> --- a/apis/novelcovid/update-checker.cronjob.yaml
> +++ b/apis/novelcovid/update-checker.cronjob.yaml
> @@ -38,12 +38,13 @@ spec:
>              - name: MAILGUN_API_KEY
>                valueFrom:
>                  secretKeyRef:
> -                  name: api-secret
> +                  name: api-secret-new
>                    key: MAILGUN_API_KEY
> +            # TODO can we get rid of CAPTCHA for this?
>              - name: CAPTCHA_SECRET
>                valueFrom:
>                  secretKeyRef:
> -                  name: api-secret
> +                  name: api-secret-new
>                    key: CAPTCHA_SECRET
>              - name: UPDATE_WEBHOOK
>                valueFrom:
> ```

The `CAPTCHA_TOKEN` is missing there. Is that added somewhere else? @dontlaugh

**Comment by dontlaugh** [comment]
good catch

File: config/index.js [hunk 2]
```diff
@@ -49,6 +49,9 @@ config.mailgunApiKey = process.env.MAILGUN_API_KEY;
 // RECAPTCHA SECRET (ONLY FOR PRODUCTION)
 config.captchaSecret = process.env.CAPTCHA_SECRET;
```
Comment (dontlaugh) [inline config/index.js:50]: @maltejur We only had `CAPTCHA_SECRET` before. I figured we would be the field we used.

Do we need both? You might consider using `captchaSecret` in your code if this refers to the same secret.

**Review by dontlaugh — COMMENTED** [review]
Just one question

File: config/index.js (hunk 2 above)
Comment (maltejur) [reply config/index.js:50]: No there are two, `CAPTCHA_SECRET` (the captcha secret, gets used by the server) and `CAPTCHA_TOKEN` (the captcha public key, gets sent to the client). I had the `CAPTCHA_TOKEN` hardcoded in the file before, but was asked to do it this way.

Comment (dontlaugh) [reply config/index.js:50]: Gotcha, I misunderstood before. I will change the K8s deployment accordingly. Standby.

**Comment by puf17640** [comment]
whoops, hope this didn't fuck something up...

**Review by ebwinters — APPROVED** [review]

**Review by puf17640 — APPROVED** [review]
lgtm

**Comment by dontlaugh** [comment]
wait for my signal to merge, updating k8s now

**Comment by dontlaugh** [comment]
Ready to roll. Merge and I'll monitor. :blush:

**Comment by puf17640** [comment]
great work guys 🙌🎉

**Comment by maltejur** [comment]
Looking good so far
aoisngokaisbgoasbngt
Can you look if that mail got added to the mailing list?

**Comment by dontlaugh** [comment]
Docs link needs to go to disease.sh

Is that configurable?

**Comment by maltejur** [comment]
"Configurable" in the sense that you can edit this line https://github.com/disease-sh/API/blob/a4add251014949591a419a213a68cbf6b3f7a004/public/index.ejs#L93
Should I open a new PR for that?

**Comment by dontlaugh** [comment]
https://github.com/disease-sh/API/issues/1006#issue-894729651

**Comment by dontlaugh** [comment]
I made a new issue to gather "new site minor bugs".

Note: All subsequent code changes should deploy automatically, without my intervention, as long as no more secrets need to change. That said, @ me if you need me

**Comment by ebwinters** [comment]
@maltejur all tweaks go in a new PR

**Pull request merged**